libraries/freertos_plus/standard/freertos_plus_posix/test/iot_test_posix_utils.c

```diff
@@ -44,14 +44,26 @@
 #include "unity_fixture.h"
 #include "unity.h"
 
+/*-----------------------------------------------------------*/
+/* Constants that represent max values for signed and unsigned definitions of time_t */
+#define TIME_T_SIGNED_MAX      ( time_t ) ( ( ( time_t ) 1 << ( ( sizeof( time_t ) << 3 ) - 1 ) ) - 1 )
```

Comment:
Why to assume that tv_sec is time_t? Is it possible to use `sizeof(signedCheck.tv_sec)` ?

Reply:
`sizeof( signedCheck.tv_sec)` would work but the `time_t` cast for the `1`  numeric constant needs to be present. I thought of using `typeof` (as `( typeof(signedCheck.tv_sec) ) ( ( ( typeof(signedCheck.tv_sec) ) 1 << ( ( sizeof( signedCheck.tv_sec ) << 3 ) - 1 ) ) - 1 )`  ) but that utility is not defined by the ISO C standard: http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf

Reply:
Also, have verified that a number of boards (Windows, NXP, Infineon )do not support the `typeof` utility.

Reply:
I think the purpose of this is to contain the max value of TIME_T and not the max value of the structure member, if the intent is the latter the name should reflect what it is.

I am not convinced that the logic is going to work all the time here. Lets test with time_t = 32 bit int.

It looks like you are doing a left-shift of 1 by 31 (which makes the intermediate result a signed int with most significant bit set, or a negative number) and then subtracting one. This seems dangerous to do, I have not tested to check the behavior but I would to know that you can explain why this is safe to do here?

Reply:
This logic has been taken from the Linux kernel implementation (`v4.7`), https://elixir.bootlin.com/linux/v4.7/source/include/linux/time.h#L11

I have tested that all the boards run successfully for this test.